test/Microsoft.AspNet.Mvc.Core.Test/Rendering/DefaultEditorTemplatesTest.cs

```diff
@@ -417,14 +439,20 @@ public void EditorFor_CallsExpectedHtmlHelper_DataTypeName(string templateName,
             viewEngine
                 .Setup(v => v.FindPartialView(It.IsAny<ActionContext>(), It.IsAny<string>()))
                 .Returns(ViewEngineResult.NotFound("", Enumerable.Empty<string>()));
+
+            var provider = new TestModelMetadataProvider();
+            provider.ForProperty<DefaultTemplatesUtilities.ObjectTemplateModel>("Property1").DisplayDetails(dd =>
+            {
+                dd.DataTypeName = templateName;
+            });
+
             var helper = DefaultTemplatesUtilities.GetHtmlHelper(
                 model,
+                null,
```

Comment:
do we need another `GetHtmlHelper` overload or two that accepts an `IModelMetadataProvider` but does not require an `IUrlHelper`?